net: openthread: manifest update with OpenThread upmerge

edmont opened the pull request:
Update manifest with latest OpenThread upmerge and handle the new required Master Key configuration.

This upmerge fixes an important bug for OpenThread MTDs not being able to keep attached to their parent: https://github.com/openthread/openthread/pull/6107

rlubos commented on samples/net/sockets/echo_client/overlay-ot.conf:
```diff
@@ -24,6 +24,7 @@ CONFIG_OPENTHREAD_L2_DEBUG=y
 CONFIG_OPENTHREAD_L2_LOG_LEVEL_INF=y
 
 CONFIG_OPENTHREAD_CHANNEL=26
+CONFIG_OPENTHREAD_MASTERKEY="00:11:22:33:44:55:66:77:88:99:aa:bb:cc:dd:ee:ff"
```
Couldn't we set this string as a default for `OPENTHREAD_MASTERKEY` option?

edmont replied:
That would be against the intention of the OpenThread PR of not having a default Master Key: https://github.com/openthread/openthread/pull/5998

rlubos replied:
Ok, understood. Please update other OT overlays that we have in the tree then:
https://github.com/zephyrproject-rtos/zephyr/blob/master/samples/net/lwm2m_client/overlay-ot.conf
https://github.com/zephyrproject-rtos/zephyr/blob/master/samples/subsys/mgmt/updatehub/overlay-ot.conf

edmont replied:
Thanks for reminding about the other samples. The FOTA one won't require the Master Key since it's using Commissioning.

rlubos commented on subsys/net/l2/openthread/Kconfig.thread:
```diff
@@ -24,6 +24,12 @@ config OPENTHREAD_XPANID
 	  Extended PAN ID for OpenThread with
 	  format "de:ad:00:be:ef:00:ca:fe"
 
+config OPENTHREAD_MASTERKEY
```
Aren't we missing new options from https://github.com/zephyrproject-rtos/openthread/pull/35/commits/28df62ec3e46471ed562cbf4a05d6dac063e2aa9 here? I.e. `CONFIG_OPENTHREAD_SRP_CLIENT` and `CONFIG_OPENTHREAD_SRP_SERVER`

edmont replied:
I was sure to have added those. Will update, thanks.

rlubos commented on subsys/net/l2/openthread/openthread.c:
```diff
@@ -374,12 +380,19 @@ int openthread_start(struct openthread_context *ot_context)
+
+		if (strlen(OT_MASTERKEY)) {
+			net_bytes_from_str(masterkey.m8, OT_MASTER_KEY_SIZE,
+							   (char *)OT_MASTERKEY);
```
The alignment seems off here.

edmont committed:
net: openthread: configure required masterkey

After the latest upmerge, OpenThread requires explicit configuration
of the Master Key. This commit adds a Kconfig symbol that can be
used to setup its value. By default no Master Key is configured and
OpenThread generates a random one.

The Sockets Echo samples are configured with a fixed key with this
commit.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

rlubos reviewed (approved):
Thanks, LGTM

edmont committed:
net: openthread: add SRP Kconfig options

Enable new features from the upmerge:
- OPENTHREAD_SRP_CLIENT
- OPENTHREAD_SRP_SERVER

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>

rlubos commented on west.yml:
```diff
@@ -112,7 +112,7 @@ manifest:
       path: modules/lib/loramac-node
     - name: openthread
-      revision: b7433ff4ef9e0d8348bdc605ff82efb9c356b233
+      revision: pull/35/head
```
Please update SHA: `aad6518ee30d3840411b21437d396f0405e947c3`

edmont committed:
manifest: openthread update with latest upmerge

Update manifest with OpenThread up to commit
'f7825b96'.

It also includes commit fixing Coprocessor
Vendor Hooks Kconfig usage in CMake.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>
